OpenDreamRuntime/Objects/MetaObjects/DreamMetaObjectMovable.cs

```diff
@@ -24,48 +24,78 @@ public DreamMetaObjectMovable(DreamRuntime runtime)
-                    DreamObject oldLoc = oldVariableValue.GetValueAsDreamObjectOfType(DreamPath.Atom);
-                    DreamList oldLocContents = oldLoc.GetVariable("contents").GetValueAsDreamList();
+                        oldLocContents.RemoveValue(new DreamValue(dreamObject));
+                    }
 
-                    oldLocContents.RemoveValue(new DreamValue(dreamObject));
-                }
+                    if (variableValue.Value != null) {
+                        DreamObject newLoc = variableValue.GetValueAsDreamObjectOfType(DreamPath.Atom);
+                        DreamList newLocContents = newLoc.GetVariable("contents").GetValueAsDreamList();
 
-                if (variableValue.Value != null) {
-                    DreamObject newLoc = variableValue.GetValueAsDreamObjectOfType(DreamPath.Atom);
-                    DreamList newLocContents = newLoc.GetVariable("contents").GetValueAsDreamList();
+                        newLocContents.AddValue(new DreamValue(dreamObject));
+                    }
 
-                    newLocContents.AddValue(new DreamValue(dreamObject));
+                    break;
                 }
-            } else if (variableName == "screen_loc") {
-                UpdateScreenLocation(dreamObject, variableValue);
+                case "screen_loc":
+                    UpdateScreenLocation(dreamObject, variableValue);
+                    break;
             }
         }
 
-        public override DreamValue OnVariableGet(DreamObject dreamObject, string variableName, DreamValue variableValue) {
-            if (variableName == "x" || variableName == "y" || variableName == "z") {
-                DreamObject location = dreamObject.GetVariable("loc").GetValueAsDreamObject();
-
-                if (location != null) {
-                    return location.GetVariable(variableName);
-                } else {
-                    return new DreamValue(0);
+        public override DreamValue OnVariableGet(DreamObject dreamObject, string variableName, DreamValue variableValue)
+        {
+            switch (variableName)
+            {
+                case "x":
+                case "y":
+                case "z":
+                {
+                    DreamObject location = dreamObject.GetVariable("loc").GetValueAsDreamObject();
+
+                    if (location != null)
+                    {
+                        return location.GetVariable(variableName);
+                    }
+                    else
+                    {
+                        return new DreamValue(0);
+                    }
                 }
-            } else {
-                return base.OnVariableGet(dreamObject, variableName, variableValue);
+                case "locs":
+                {
+                    //TODO multi-tile support
+                    var list = DreamList.Create(Runtime);
+                    list.AddValue(dreamObject.GetVariable("loc"));
+                    return new DreamValue(list);
```

Comment:
Why do we need to wrap the list in a new dreamvalue? Isn't a list already one?

Reply:
A list is a DreamObject, which has to be wrapped by a DreamValue